### ISSUE TITLE
Add version number to simdjson.h/.cpp 

### ISSUE THREAD

**Issue by dan-ryan** [simdjson/simdjson#1339]
I'm just trying out Simdjson and downloaded [simdjson.h](https://github.com/simdjson/simdjson/blob/master/singleheader/simdjson.h)/[.cpp](https://github.com/simdjson/simdjson/blob/master/singleheader/simdjson.cpp) but I noticed it has no version number in the comments. I believe that would be helpful to have.

Another thing is that the Quick Start section for those files is pointing to Master. Maybe it should point to the most recent tagged release version?

**Comment by TkTech** [comment]
The version is in the header as SIMDJSON_VERSION.

**Comment by lemire** [comment]
@TkTech is correct...

```C++
/** The version of simdjson being used (major.minor.revision) */
#define SIMDJSON_VERSION 0.7.0

namespace simdjson {
enum {
  /**
   * The major version (MAJOR.minor.revision) of simdjson being used.
   */
  SIMDJSON_VERSION_MAJOR = 0,
  /**
   * The minor version (major.MINOR.revision) of simdjson being used.
   */
  SIMDJSON_VERSION_MINOR = 7,
  /**
   * The revision (major.minor.REVISION) of simdjson being used.
   */
  SIMDJSON_VERSION_REVISION = 0
};
} // namespace simdjson
```

> Another thing is that the Quick Start section for those files is pointing to Master. Maybe it should point to the most recent tagged release version?

It should.

**Comment by lemire** [comment]
> It should.

It should except that I do not want to overburden the release process by requiring too many updates to the documentation.

We point at the `singleheader` files which are rarely updated.

**Comment by dan-ryan** [comment]
> The version is in the header as SIMDJSON_VERSION.
Oh yep. But that is on line 2018. It would be helpful it if was at the top of the file that way those who don't know about this library know exactly which version is being used. Otherwise, I can just manually do it :)

> It should except that I do not want to overburden the release process by requiring too many updates to the documentation.
Understood. Just that people might be using a master version that could have issues instead of a stable version.

**Comment by dan-ryan** [comment]
> We point at the singleheader files which are rarely updated.

Instead of updating the documentation. You could just update the Master singleheader files only when you do a release.

**Comment by lemire** [comment]
> Instead of updating the documentation. You could just update the Master singleheader files only when you do a release or something like that.

That's effectively what we do in practice but we have had to do updates of these files between releases for testing purposes so for production use, you need to rely on one of our releases and not rely on the tip of our main git branch. But the larger point is that there is simply no scenario where I would recommend grabbing files from the tip of our main branch on github for production uses. I think we make it reasonably easy to find our releases. On top of [releasing on github](https://github.com/simdjson/simdjson/releases), we also have releases via package managers such asa MSYS2, conan, vcpkg, brew, apt, FreeBSD and so forth.

If you are preparing software for production use, I recommend against browsing the comments in the source files to know which version you are using. one should work from a package manager, or grab the release archives by their tagged URLs such https://github.com/simdjson/simdjson/archive/v0.7.0.zip or use git tags. We also document how to use it as a CMake dependency.


I am not sure that our quick-start guide should be taken as a guide to preparing a production-ready system. We have this explicit statement right in our README: "If you are planning to use simdjson in a product, please work from one of our releases."

What I expect users to do is to just grab the code initially and play with it for a quick assessment. If it meets their need, then I expect them to get serious and grab a release in some standard way (documented above).

**Comment by dan-ryan** [comment]
"That's effectively what we do in practice"
Great.

"one should work from a package manager"
Yes ideally package managers are the way to go. it's a shame that the project I'm working on doesn't have one. 

"If it meets their need, then I expect them to get serious and grab a release in some standard way"
In a perfect world :) Anyway, feel free to ignore this suggestion. I'll just manually add in a version number for now.

**Comment by lemire** [comment]
@dan-ryan I do not think that adding a version tag as a comment at the top of the singleheader files is a bad idea. It could prove useful. Would you prepare a pull request modifying our setup and scripts that implements the desired feature?

**Comment by dan-ryan** [comment]
When I have some free I'll see if I can have a look. Thanks.
